test/packages/good_v2/data_stream/routing_rules/routing_rules.yml

```diff
@@ -0,0 +1,35 @@
+routing_rules:
+  # "Local" routing rules are included under this current dataset, not a special case
+  - dataset: routing_rules
+    rules: # Route error logs to `routing_rules.error` when they're sourced from an error logfile
+    - dataset: routing_rules.error
+      if: "ctx?.file?.path?.contains('/var/log/routing_rules/error')"
+      namespace:
+        - "{{labels.data_stream.namespace}}"
+        - default
+
+    # Route access logs to `routing_rules.access` when they're sourced from an access logfile
+    - dataset: routing_rules.access
+      if: "ctx?.file?.path?.contains('/var/log/routing_rules/access')"
+      namespace:
+        - "{{labels.data_stream.namespace}}"
+        - default
+
+  # Route K8's container logs to this catch-all dataset for further routing
+  - dataset: k8s.router
```

Comment:
We cannot make this a map because of support for dots, right?

Reply:
Exactly, I tried to set a map for these definitions. However, that `dot` causes that the keys are expanded and therefore it is interpreted as an object.

Here it is what I tried https://github.com/elastic/package-spec/pull/535#discussion_r1229428169